src/wasm-type.h

```diff
@@ -303,6 +308,112 @@ struct Array {
   std::string toString() const;
 };
 
+struct Rtt {
+  enum Kind {
+    // Basic heap types
+    FuncKind,
+    ExternKind,
+    AnyKind,
+    EqKind,
+    I31Kind,
+    _last_basic_kind = I31Kind,
+    // Compound heap types
+    SignatureKind,
+    StructKind,
+    ArrayKind,
+  } kind;
```

Comment:
Since we don't have a concept of `HeapType`, this encodes both basic and compound heap types using a single enum. Is that fine or should we introduce `HeapType` for clarity (doesn't give us much functionally I think)?

Reply:
I don't think we need to add it as a separate concept, unless you feel that it would make development easier by making things more similar to the spec. I'll leave that up to you as the one writing the code :)